### DISCUSSION TITLE
when selecting, please allow choose number of files being selected

### DISCUSSION THREAD

**Discussion by edmundlaugasson** [doublecmd/doublecmd#1128] [Suggestions]
### Description

Sometimes there is needed to split the content of folder into certain amount of files.

### Proposed technical implementation details (optional)

On "Define template" [tab "Advanced"](https://doublecmd.github.io/doc/en/findfiles.html#advanced) there could be in addition of current age, size, date be also additional option: number of files. Defining number of files there could be parameters like: from beginning, from end, starting position from beginning, starting position from end, custom selecting pattern. Under custom pattern there could be: select x number of files, then not select next x number of files, then again select x number of files. Also selection invert feature for selection would be appreciated.
Tried to use [regular expressions](https://doublecmd.github.io/doc/en/regexp.html) but could not figure out, how to select first 598 files among 1196 ones. Ended up with manual selection...

**Comment by Skif-off** [comment]
> Proposed technical implementation details (optional)
> On "Define template" [tab "Advanced"](https://doublecmd.github.io/doc/en/findfiles.html#advanced) there could be in addition of current age, size, date be also additional option: number of files.

Why? Why not [Select group](https://doublecmd.github.io/doc/en/images/imgDC/pic27.png)? For example, with additional inputbox ("Number:") and checkbox ("From the current position")
(The list of files returned by "Find files" is not necessarily the same as what you see in the file panel).

> Sometimes there is needed to split the content of folder into certain amount of files.

Do you mean just the current directory without the flat view?
If true, what about [temporary] workaround with Lua?

**Comment by edmundlaugasson** [comment]
Could be Select group but don't find that number input box and from current position checkbox.
Don't understand, what you mean under flat view. Also not familiar with Lua. I proposed UI design to solve selecting issues.

**Comment by Skif-off** [comment]
> Could be Select group but don't find that number input box and from current position checkbox.

I just proposed a more adequate (IMHO) place for this feature :)

> Don't understand, what you mean under flat view.

See [cm_FlatView/cm_FlatViewSel](https://doublecmd.github.io/doc/en/cmds.html#cm_FlatView).


About Lua,
<details>
<summary>I mean something like this</summary>

```lua
-- (cross-platform)
--
--[[
Select N file in the active file panel.
Specify "N" to select from the first, or "-N" to select from the current cursor position.

How to use:
internal command: cm_ExecuteScript
params:
  path-to-script
  %"0%f0
]]

local params = {...}

-- Get the script name for window titles.
local sScrName = debug.getinfo(1).source
if string.sub(sScrName, 1, 1) == '@' then
  sScrName = string.sub(sScrName, 2, -1)
end
sScrName = SysUtils.ExtractFileName(sScrName)

-- Check parameters
if #params ~= 1 then
  Dialogs.MessageBox("Check the number of parameters!", sScrName, 0x0030)
  return
end

local bBegin = true
local iBegin = 1

-- Get a number
local bAct, sNum = Dialogs.InputQuery(sScrName, "Number of files:", false, "0")
if bAct == false then return end
if string.sub(sNum, 1, 1) == "-" then
  bBegin = false
  sNum = string.sub(sNum, 2, -1)
end
local iNum = tonumber(sNum)
if iNum == 0 then return end

-- Get a list of files
local aList = {}
local iCount = 1
local sTmpFile = os.tmpname()

DC.ExecuteCommand("cm_MarkMarkAll", "attr=")
DC.ExecuteCommand("cm_SaveSelectionToFile", sTmpFile)
DC.ExecuteCommand("cm_MarkUnmarkAll", "attr=")

local h, err = io.open(sTmpFile, "r")
if h == nil then
  Dialogs.MessageBox(sTmpFile .. "\nError: " .. err, sScrName, 0x0030)
  return
end
for line in h:lines() do
  aList[iCount] = line
  iCount = iCount + 1
end
h:close()
os.remove(sTmpFile)
if #aList == 0 then return end

-- Get the starting position
if bBegin == false then
  for i = 1, #aList do
    if aList[i] == params[1] then
      iBegin = i
      break
    end
  end
end

-- Check the upper bound
local iEnd = iBegin + iNum - 1
if iEnd > #aList then
  local iButton = Dialogs.MessageBox("The number " .. iNum .. " is too big. Just ignore it?", sScrName, 0x0030 + 0x0001)
  if iButton == 0x0001 then
    iEnd = #aList
  else
    return
  end
end

-- Select files
h, err = io.open(sTmpFile, "wb")
if h == nil then
  Dialogs.MessageBox(sTmpFile .. "\nError: " .. err, sScrName, 0x0030)
  return
end
h:write(table.concat(aList, "\n", iBegin, iEnd))
h:close()
DC.ExecuteCommand("cm_LoadSelectionFromFile", sTmpFile)
os.remove(sTmpFile)

```
Save as, for example `test.lua `, add button (see comment in the beginning of file) and try to enter "5" and "-5" (without quotes).

If you want to work with files only, then find
```lua
DC.ExecuteCommand("cm_MarkMarkAll", "attr=")
```
and try to replace on
```lua
DC.ExecuteCommand("cm_MarkMarkAll", "attr=d-")
```
</details>

**Comment by edmundlaugasson** [comment]
> Why not [Select group](https://doublecmd.github.io/doc/en/images/imgDC/pic27.png)? For example, with additional inputbox ("Number:") and checkbox ("From the current position")

Could be also directly in search box. Would be easier, that's true. Understood, that fine tuned searches are under that _Define..._ button.

> See [cm_FlatView/cm_FlatViewSel](https://doublecmd.github.io/doc/en/cmds.html#cm_FlatView).

I guess the current directory splitting mode should be selectable: either without (by default) or with flat view. This flat view seems to open folder and when use flat view then this seems already recursively (including subdirectories). The need split current directory content recursively depends very much of current need. My current need was just current directory as there was no subdirectories but these might be and people's needs are different.
